indexing-service/src/main/java/io/druid/indexing/overlord/http/OverlordResource.java

```diff
@@ -350,24 +350,12 @@ public Response shutdownTasksForDataSource(@PathParam("dataSource") final String
           public Response apply(TaskQueue taskQueue)
           {
             final List<Task> tasks = taskStorageQueryAdapter.getActiveTasks();
-            boolean ownTask = false;
             for (final Task task : tasks) {
               if (task.getDataSource().equals(dataSource)) {
                 taskQueue.shutdown(task.getId());
-                ownTask = true;
               }
             }
```

Comment:
You can also use this API `getActiveTaskInfo(@Nullable String dataSource)` which takes a `dataSource` as argument and filters at the DB query time, so you will only get the matching `dataSource` active tasks. So like below. It might be efficient as you don't have to compare every task's dataSource in the for loop.
```
final List<TaskInfo<Task, TaskStatus>> tasks = taskStorageQueryAdapter.getActiveTaskInfo(dataSource);
            for (final TaskInfo<Task, TaskStatus> task : tasks) {
              taskQueue.shutdown(task.getId());
            }```

Reply:
Cool, fixed!